### DISCUSSION TITLE
Startup time crud service

### DISCUSSION THREAD

**Discussion by giovanni-motterle-mia-care** [mia-platform/community#431] [Q&A]
### What product are you having troubles with?

Marketplace

### What Console version are you using?

1.96.3

### Description

Hi, 
I'm using crud-service (version 6.9.1).
I noticed that whenever I need to restart the service (e.g. when I add a new field to a collection), the service takes about 5 minutes to become up and running.
![Screenshot 2024-03-11 122936](https://github.com/mia-platform/community/assets/110187011/0879c938-4808-4e39-9095-230d2e63827d)
![Screenshot 2024-03-11 122540](https://github.com/mia-platform/community/assets/110187011/d3a97cb4-73d3-4f75-ac7c-83e52a8533b9)
Above you can see that at 12:05:45 it prints "Server listening ..." but then it continues with the startup process (so I guess it's not really listening).
Then, at 12:05:47 it logs the last info about start-up and goes silent for about 4 minutes (although calls are still coming and are redirected to the other crud-service pod that's still running).
I have about 140 collections and 13 views, the service is configured with 100Mi req and 1000Mi limit for Memory and 100m and 1000m for CPU. Despite the first minute of CPU reaching the limit, it then stays below request.

How can I understand what's going on in those 4 minutes?

Thanks

**Comment by fredmaggiowski** [comment]
Hi, the first releases of the CRUD Service v6.9.x have known performance issues at boot, I'd try upgrading to the latest available patch (for sure [>6.9.4](https://github.com/mia-platform/crud-service/releases/tag/v6.9.4)), or even 6.10.0, to see if this fixes this behavior in your use-case

**Reply by giovanni-motterle-mia-care** [reply]
Ok thanks, I will let you know the outcome.

**Reply by giovanni-motterle-mia-care** [reply]
Hi, I'm now using crud-service:6.9.4, we added about 80 new collections to the ones that were already there when I opened the thread (now I have about 150 collections).
If I try to release, the service is not going up at all.
```
Events:
  Type    Reason     Age    From               Message
  ----    ------     ----   ----               -------
  Normal  Scheduled  5m16s  default-scheduler  Successfully assigned ia-backend-development/crud-service-56d9c9c9b7-g42rj to gke-noprod-application-nodes-702ab26d-z0cf
  Normal  Pulling    5m16s  kubelet            Pulling image "docker.io/miaplatform/crud-service:6.9.4"
  Normal  Pulled     5m15s  kubelet            Successfully pulled image "docker.io/miaplatform/crud-service:6.9.4" in 594.872112ms (594.890505ms including waiting)
  Normal  Created    5m15s  kubelet            Created container crud-service
  Normal  Started    5m15s  kubelet            Started container crud-service
```
Although above is present "Started container crud-service", in console the pod stays in "Warning".
Here the configuration of resources, probes and resources consumption
![328797763-2cddcdb4-1127-4849-bc21-3181a5f6d231](https://github.com/mia-platform/community/assets/110187011/c28188a1-95cf-4d21-83d5-a86d8066cf3d)
![328797767-7fee7132-5abb-4c6b-b366-4ade74be9b09](https://github.com/mia-platform/community/assets/110187011/f1bff7f5-d892-440c-93f4-fa31e10de8b5)
![328797769-02254918-a14c-453f-b7e6-f9a3b28442e6](https://github.com/mia-platform/community/assets/110187011/9ee9190f-c98b-40d2-ac54-e1fc0559c592)

I tried deleting all views (14) and then the service has no problems, what do you think I can check next?

**Reply by fredmaggiowski** [reply]
What is the reason the pod isn't starting? Is it being killed by Kubernetes for not matching the liveness probe in time? If so I'd add the startup probe configuration.

Since it appears that views are the ones affecting startup performances I'd consider looking up into the code if there is some optimization that can be done as well (@danibix95 FYI)

**Reply by danibix95** [reply]
Hi, are the views standard or "writable" (the ones that support lookup edits)?

**Reply by giovanni-motterle-mia-care** [reply]
Hi, yes the reason appears to be kubernetes killing the pod after the limit of failures for readiness is reached. I will try to configure one.

The views are all with "Writable View:" toggle off.
It's kind of an odd situation because first we added a bunch of new collections and the pod was no more going up, then we removed all the views (not the new collections) and it went up quickly.
The current configuration that is running ok (with very slow startup times) is (with respect the starting point) with half of the new collections and a bunch of new views.
It's not crystal clear to me if the problem are only views or a combination of a lot of collections and some view.
We can look into it on a call if necessary.
